src/sql-editor/components/SQLEditor.tsx

```diff
@@ -156,7 +153,7 @@ const resolveLanguage = (monaco: Monaco, languageDefinitionProp: LanguageDefinit
     if (!custom) {
       throw Error(`Unknown Monaco language ${languageDefinitionProp?.id}`);
     }
-    return custom;
+    return { completionProvider: getStandardSQLCompletionProvider, ...custom, ...languageDefinitionProp };
```

Comment:
I'm kind of lost here. In your example you merge the `getStandardSQLCompletionProvider` in the custom completion provider but in this part of the code it will also merge it again? Can you please shed some light? Sorry but the extensibility is not crystal clear to me.

```
export const getRedshiftCompletionProvider: (args: CompletionProviderGetterArgs) => LanguageCompletionProvider =
  ({ getColumns, getTables }) =>
  (monaco, language) => {
    return {
      // get standard SQL completion provider which will resolve functions and macros
      ...(language && getStandardSQLCompletionProvider(monaco, language)),
      tables: {
        resolve: async () => getTables.current(),
      },
      columns: {
        resolve: async (t: string) => getColumns.current(t),
      },
    };
  };
```

Reply:
Good question! The redshift example is in the case a custom language is provided. The if statement on line 156 in the SQLEditor.tsx is when a language from the monaco registry is specified. In that case if will use a custom completion item provider if it was specified, and if not it will fallback to the standard one. Hope that answers your question!